coalib/bearlib/aspects/__init__.py

```diff
@@ -0,0 +1,151 @@
+"""
+This package contains all aspects that bears can detect or fix and their
+documentation.
+"""
+
+from inspect import cleandoc
+
+from coala_utils.decorators import classproperty, enforce_signature
+from coalib.bearlib.naming_conventions import to_snakecase
+
+
+class ConsistencyCheckMixin:
+    """
+    A mixin that can be inherited from and offers a ``check_consistency``
+    method.
+    """
+
+    def check_consistency(self):
+        """
+        Checks the consistency of the self object. Consistency is given, if all
+        members of the object evaluate to ``True``.
+
+        :return: ``True`` or ``False``
+        """
+        return all(getattr(self, key)
+                   for key in dir(self)
+                   if not key.startswith('_'))
+
+
+class AspectDocumentation(ConsistencyCheckMixin):
+    """
+    This class contains documentation about an aspect described by the Aspect
+    class.
+
+    The documentation is consistent if all members are given:
+
+    >>> AspectDocumentation('defined', '', '', '').check_consistency()
+    False
+    >>> AspectDocumentation('definition', 'cause', 'example',
+    ...                     'example_language', 'importance',
+    ...                     'fix').check_consistency()
+    True
+    """
+
+    @enforce_signature
+    def __init__(self, definition: str='', cause: str='', example: str='',
+                 example_language: str='', importance_reason: str='',
+                 fix_suggestions: str=''):
+        """
+        Contains documentation for an aspect.
+
+        :param definition:        What is this about?
+        :param cause:             Information on how such a problem can happen.
+        :param example:           An example in a well known language.
+        :param example_language:  The language used for the example.
+        :param importance_reason: A reason why this aspect is important.
+        :param fix_suggestions:   Suggestions on how this can be fixed.
+        """
+        super().__init__()
+
+        self.definition = cleandoc(definition)
+        self.cause = cleandoc(cause)
+        self.example = cleandoc(example)
+        self.example_language = cleandoc(example_language)
+        self.importance_reason = cleandoc(importance_reason)
+        self.fix_suggestions = cleandoc(fix_suggestions)
+
+
+class Aspect:
```

Comment:
i feel like the flow of this file is not good. shouldnt it first be the Aspect class in the file and then its documentation? i started reading about its documentation without understanding anything, finally coming to the Aspect class somewhere at the end of the file ..